### ISSUE TITLE
Signature verification fails due to different output values

### ISSUE THREAD

**Issue by devrandom** [rust-bitcoin/rust-lightning#888]
I am sending dust HTLCs in a loop, and the channel force-closes sometimes due to signature mismatch.

The signed transactions differ in value to-holder by one payment worth.

**Comment by devrandom** [comment]
- [alice.txt](https://github.com/rust-bitcoin/rust-lightning/files/6342502/alice.txt)
- [bob.txt](https://github.com/rust-bitcoin/rust-lightning/files/6342503/bob.txt)

**Comment by ksedgwic** [comment]
```
  alice: 2021-04-20 9:09:07 TRACE [lightning::ln::channel:967]    ...not including outbound HTLC 130 (hash d14e3d762a9b4600bb983ef3afc08d7eae6cb1c2e120fe8bcbc9ff25536234a7) with value 10000 due to state (AwaitingRemovedRemoteRevoke)
  bob:   2021-04-20 9:09:07 TRACE [lightning::ln::channel:937]    ...including inbound LocalRemoved dust HTLC 130 (hash d14e3d762a9b4600bb983ef3afc08d7eae6cb1c2e120fe8bcbc9ff25536234a7) with value 10000
```

**Comment by TheBlueMatt** [comment]
Closed by #891.
